### ISSUE TITLE
I2I: [fetch-error] and [diffable] in amp-list

### ISSUE THREAD

**Issue by choumx** [ampproject/amphtml#23465]
## Summary

- Supports "fetch error" UI via `[fetch-error]` child (#22704)
- Shippable path for DOM diffing via optional `[diffable]` (experiment added in #17000)

These features are distinct but can interact: `[diffable]` allows "fetch error" UI to modify the position of the actual content to avoid overlapping. More details below.

### Fetch error 
A "fetch error" UI is a nice way to inform the user that the initial content is stale. The proposed format follows amp-form's `[submit-error]` pattern:

```css
amp-list [fetch-error] { display: none; }
amp-list.i-amphtml-list-fetch-error [fetch-error] { display: block; }
```

```html
<amp-list src="...">
  <div fetch-error>
    Failed to fetch latest content!
  </div>
</amp-list>
```

### DOM diffing
Reduces flickering when transitioning from initial to rendered content, or re-rendering amp-list via amp-bind. E.g. don't replace `amp-img` if it didn't change.

Other than enabling DOM diffing, `[diffable]` allows specifying initial content without using `[placeholder]` (which is absolute-positioned).

This means initial content can be diff'ed against rendered content, and initial content's position can be offset by fetch error UI.

```html
<amp-list diffable>
  <!-- Initial content.  -->
  <div role=list>
    <div role=listitem>Initial content will be diff'ed</div>
    <div role=listitem><amp-img src="..."></amp-img></div>
  </div>
</amp-list>
```

Diffing is done by [setDOM](https://github.com/DylanPiercey/set-dom). AMP elements are skipped and diff'ed manually.

/cc @cathyxz @ampproject/wg-ui-and-a11y

**Comment by choumx** [comment]
Draft implementation: #23470
